portals/publisher/src/main/webapp/source/src/app/data/api.js

```diff
@@ -392,6 +392,71 @@ class API extends Resource {
         });
     }
 
+    saveAPIDesignAssistant() {
+        const promisedAPIResponse = this.client.then(client => {
+            const properties = client.spec.components.schemas.API.properties;
+            const data = {};
+            Object.keys(this).forEach(apiAttribute => {
+                if (apiAttribute in properties) {
+                    data[apiAttribute] = this[apiAttribute];
+                }
+            });
+            const payload = {
+                'Content-Type': 'application/json'
+            };
+            const requestBody = {
+                'requestBody': data,
+            };
+            return client.apis['APIs'].createAPI(payload, requestBody, this._requestMetaData());
+        });
+        return promisedAPIResponse.then(response => {
+            return new API(response.body);
+        });
+    }
+
+    sendChatAPIDesignAssistant(query, sessionId) {
+        return this.client.then(client => {
+            const data = {
```

Comment:
Let's not define the data variable here. Instead define the content within the requestBody obj itslef.